src/components/forms/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,52 @@
+import './SignUpForm.sass';
+import {useForm} from 'react-hook-form';
+import BasicTextField from '@/components/inputs/BasicTextField/BasicTextField';
+import BasicBtn from '@/components/btns/BasicBtn/BasicBtn';
+import {signUpSchema} from '@/constants/signUp.validation';
+import {yupResolver} from '@hookform/resolvers/yup';
+
+interface FormData {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  phoneNumber?: string;
+}
+
+const SignUpForm: React.FC = () => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: {errors}
+  } = useForm<FormData>({
+    resolver: yupResolver(signUpSchema)
+  });
+
+  const onSubmit = (data: FormData): void => {
+    console.log(data);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <div className="signUpForm__input">
+        <BasicTextField name="firstName" control={control} label="First name" />
+      </div>
+      <div className="signUpForm__input">
+        <BasicTextField name="lastName" control={control} label="Last name" />
+      </div>
```

Comment:
You can put this div inside you component and pass class as a prop directly. The markup will be much better.